### PR TITLE
Update dotnet nuget why docs with assets file info

### LINKED ISSUE #13623: Update dotnet nuget why docs with assets file info
When this is complete:
* https://github.com/NuGet/Home/issues/13576

We need to update the docs for dotnet nuget why so that we can explain that non-sdk projects can restore with `msbuild -t:restore`, then use `msbuild path\to\project.csproj -getProperty:ProjectAssetsFile`, and then pass that assets file path to `dotnet nuget why`.

### PR DESCRIPTION
**Pull request by zivkan**
## Summary

`dotnet nuget why` is a useful tool to understand the project's package graph. This is particularly true when dealing with transitive packages with known vulnerabilities, as described in this blog post: https://devblogs.microsoft.com/nuget/nugetaudit-2-0-elevating-security-and-trust-in-package-management/

Fixes: https://github.com/NuGet/Home/issues/13623

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/tools/dotnet-nuget-why.md](https://github.com/dotnet/docs/blob/a23148249271792dcbab9e798887d4217cb8fb58/docs/core/tools/dotnet-nuget-why.md) | [docs/core/tools/dotnet-nuget-why](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-nuget-why?branch=pr-en-us-42101) |

### CODE REVIEW

**Comment by tdykstra** [comment]
Waiting for review by @dotnet/nuget-team

**Review by Nigusu-Allehu — APPROVED** [review]

**Review by nkolev92 — APPROVED** [review]

**Review by tdykstra — APPROVED** [review]

**Pull request merged**
